activerecord/test/cases/insert_all_test.rb

```diff
@@ -142,4 +143,25 @@ def test_insert_all_raises_on_unknown_attribute
       Book.insert_all! [{ unknown_attribute: "Test" }]
     end
   end
+
+  def test_insert_all_on_relation
+    skip unless supports_insert_on_duplicate_skip?
+
+    author = Author.create!(name: "Jimmy")
+
+    assert_difference "author.books.count", +1 do
+      author.books.insert_all([{ name: "My little book", isbn: "1974522598" }])
+    end
+  end
+
+  def test_insert_all_on_relation_precedence
+    skip unless supports_insert_on_duplicate_skip?
+
+    author = Author.create!(name: "Jimmy")
+    second_author = Author.create!(name: "Bob")
+
+    assert_difference "author.books.count", +1 do
+      author.books.insert_all([{ name: "My little book", isbn: "1974522598", author_id: second_author.id }])
+    end
+  end
```

Comment:
Could you add a test (or tests) to cover also `create_with`, which this PR adds support for? — loosely like:

```ruby
Book.create_with(isbn: "X").insert_all([ { name: "A" }, { name: "B" } ])
assert_equal 2, Book.where(isbn: "X").count
```

Reply:
done